Tests/UnitTests/Reporters/JunitXMLReporterTest.jsl

```diff
@@ -0,0 +1,421 @@
+));
+
+// 9.0 JunitXMLReporter one simple passing
+
+ut test( JunitXMLReporters, "JunitXMLReporter one simple passing", Expr(
+
+	with junit reporter(Expr(
+	   ut test ("Simple Pass", "One plus one equals two", Expr(
+	     ut assert that(Expr(1+1), ut equal to (2));
+	   ));
+	));
+	
+	ut assert that(Expr(reporter << successes), ut equal to ({{"Simple Pass ⮚ One plus one equals two ⮚ 1", 1 + 1, "equal to 2", Empty()}}));
+	
+	pattern = 
+	"\[<\?xml version="1.0" encoding="UTF-8" \?>
+	<testsuites failures="0" id="\d+_\d+" tests="1">
+	  <testsuite failures="0" name="Simple Pass" tests="1">
+		<testcase name="One plus one equals two_1">
+		</testcase>
+	  </testsuite>
+	</testsuites>
+	]\";
+	
+	//replace tabs and carriage returns used for easier readability
+	Substitute Into(pattern, "\!N", ".+", "\!r", ".+", "\!t", "");
+
+	match = Regex Match(report, pattern);
+	
+	ut assert that(Expr(match), ut length(1));
+	ut assert that(Expr(report), ut valid xml() );
+	ut assert that(Expr(match[1]), ut equal to(report));
+));
+
+// 10.0 JunitXMLReporter one simple failing
+
+ut test( JunitXMLReporters, "JunitXMLReporter one simple failing", Expr(
+
+	with junit reporter(Expr(
+	    ut test ("Simple Fail", "One plus one equals one", Expr(
+	       ut assert that(Expr(1+1), ut equal to (1));
+	    ));
+	));
+	
+	ut assert that(Expr(reporter << failures), ut equal to ({{"Simple Fail ⮚ One plus one equals one ⮚ 1", 1 + 1, "equal to 1", "was 2", 0, Empty()}}));
+	
+	pattern = 
+	"\[<\?xml version="1.0" encoding="UTF-8" \?>
+	<testsuites failures="1" id="\d+_\d+" tests="1">
+	  <testsuite failures="1" name="Simple Fail" tests="1">
+		<testcase name="One plus one equals one_1">
+		  <failure type="failure">
+			Expected: 1 \+ 1 equal to 1
+			But: was 2
+		  </failure>
+		</testcase>
+	  </testsuite>
+	</testsuites>
+	]\";
+	
+	//replace tabs and carriage returns used for easier readability
+	Substitute Into(pattern, "\!N", ".+", "\!r", ".+", "\!t", "");
+
+	match = Regex Match(report, pattern);
+	
+	ut assert that(Expr(Nitems(match)), ut equal to(1));
+	ut assert that(Expr(report), ut valid xml() );
+	ut assert that(Expr(match[1]), ut equal to(report));
+));
+
+// 11.0 Assertion with 4 part label works
+
+ut test(JunitXMLReporters, "Assertion with 4 part label works", Expr(
+	with junit reporter(Expr(
+		ut test("A", "a", Expr(
+			ut assert value(1, 0, "extra info");
+		));
+	));
+	ut assert that( Expr(report), ut valid xml() );
+	
+	pattern = "<\?xml version=\!"1.0\!" encoding=\!"UTF-8\!" \?>
+	<testsuites failures=\!"1\!" id=\!"\d+_\d+\!" tests=\!"1\!">
+	  <testsuite failures=\!"1\!" name=\!"A\!" tests=\!"1\!">
+		<testcase name=\!"a_extra info_1\!">
+		  <failure type=\!"failure\!">
+			Expected: Expr\(1\) equal to 0
+			But: was 1
+		  </failure>
+		</testcase>
+	  </testsuite>
+	</testsuites>
+	";
+
+	//replace tabs and carriage returns used for easier readability
+	Substitute Into(pattern, "\!r", ".+", "\!t", "");
```

Comment:
This test is failing when run from `RunTests.jsl`. It needs the \!N replacement.

Reply:
fixed in commit 7dda28f6a337f12e2526a4b768e7eab2f21e3201